app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -85,7 +85,10 @@ class RecentSyncedTabFeature(
     override fun stopLoading() = Unit
 
     override fun onError(error: SyncedTabsView.ErrorType) {
-        store.dispatch(AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.None))
+        val isSyncing = store.state.recentSyncedTabState == RecentSyncedTabState.Loading
+        if ((!isSyncing && error == SyncedTabsView.ErrorType.NO_TABS_AVAILABLE) || error.isFatal()) {
```

Comment:
Since we do have a RecentSyncedTabFeatureTest, be sure to add or update the tests as necessary.